test.ps1

```diff
@@ -25,89 +25,100 @@ param (
-$sourceDir = Resolve-Path "exercises"
 
-function Configlet-Lint {
+function Invoke-Configlet-Lint {
     Write-Output "Linting config.json"
-    Run-Command "./bin/fetch-configlet"
-    Run-Command "./bin/configlet lint"
+    Invoke-CommandExecution "./bin/fetch-configlet"
+    Invoke-CommandExecution "./bin/configlet lint"
 }
 
-function Build-Generators { 
+function Invoke-DotNetBuild-Generators { 
     Write-Output "Building generators"
-    Run-Command "dotnet build ./generators"
+    Invoke-CommandExecution "dotnet build ./generators"
 }
 
-function Clean {
+function Clean($BuildDir) {
     Write-Output "Cleaning previous build"
-    Remove-Item -Recurse -Force $buildDir -ErrorAction Ignore
+    Remove-Item -Recurse -Force $BuildDir -ErrorAction Ignore
 }
-
-function Copy-Exercises {
+function Copy-Exercise($SourceDir, $BuildDir) {
     Write-Output "Copying exercises"
-    Copy-Item $sourceDir -Destination $buildDir -Recurse
+    Copy-Item $SourceDir -Destination $BuildDir -Recurse
 }
 
-function Enable-All-Tests {
+function ConvertTo-UnSkippedTest($BuildDir) {
     Write-Output "Enabling all tests"
-    Get-ChildItem -Path $buildDir -Include "*Tests.cs" -Recurse | ForEach-Object {
+    Get-ChildItem -Path $BuildDir -Include "*Tests.cs" -Recurse | ForEach-Object {
         (Get-Content $_.FullName) -replace "Skip = ""Remove this Skip property to run this test""", "" | Set-Content $_.FullName
     }
 }
 
-function Test-Refactoring-Projects {
-    Write-Output "Testing refactoring projects"
-    @("tree-building", "ledger", "markdown") | ForEach-Object { Run-Command "dotnet test $practiceExercisesDir/$_" }
+function Test-ExeciseWithRefactoring($PracticeExercisesDir) {
+    Write-Output "Testing refactoring exercises"
+    @("tree-building", "ledger", "markdown") | ForEach-Object { Invoke-CommandExecution "dotnet test $practiceExercisesDir/$_" }
 }
 
-function Replace-Stubs-With-File ($ExercisesDir, $ReplaceFileName) {
-    Get-ChildItem -Path $ExercisesDir -Include "*.csproj" -Recurse | ForEach-Object {
-        $stub = Join-Path $_.Directory ($_.BaseName + ".cs")
-        $example = Join-Path $_.Directory ".meta" $ReplaceFileName
-    
-        Move-Item -Path $example -Destination $stub -Force
+function Set-SolutionFileForExercise {
+    [CmdletBinding(SupportsShouldProcess)]
+    param($ExercisesDir, $ReplaceFileName)
+
+    if ($PSCmdlet.ShouldProcess($ReplaceFileName)) {
+        Get-ChildItem -Path $ExercisesDir -Include "*.csproj" -Recurse | ForEach-Object {
+            $stub = Join-Path -Path $_.Directory ($_.BaseName + ".cs")
+            $example = Join-Path -Path $_.Directory ".meta" $ReplaceFileName
+        
+            Move-Item -Path $example -Destination $stub -Force
+        }
     }
 }
 
-function Replace-Stubs {
-    Write-Output "Replacing concept exercise stubs with exemplar"
-    Replace-Stubs-With-File $conceptExercisesDir "Exemplar.cs"
+function Set-AllSolutionsForExercise {
```

Comment:
Why this naming change?

Reply:
Replace-Stubs had two warning:
- Replace: PSUseApprovedVerbs - The cmdlet 'Replace-Stubs' uses an unapproved verb.
- Stubs: PSUseSingularNouns  - The cmdlet 'Replace-Stubs' uses a plural noun. A singular noun should be used instead.

Reply:
> The cmdlet 'Replace-Stubs' uses an unapproved verb.

Oh my, I think that is a very harsh rule. How could they possible have a whitelist of all valid verbs?

How about we rename to `Use-ExampleImplementation`? Is that valid?

Reply:
https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.1

yes :)